account-kit/signer/src/apiKeySigner.ts

```diff
@@ -0,0 +1,246 @@
+import { type ConnectionConfig, type SmartAccountSigner } from "@aa-sdk/core";
+import { ApiKeyStamper } from "@turnkey/api-key-stamper";
+import type {
+  AlchemySignerClientEvents,
+  AuthenticatingEventMetadata,
+  CreateAccountParams,
+  EmailAuthParams,
+  EnableMfaParams,
+  EnableMfaResult,
+  GetWebAuthnAttestationResult,
+  MfaFactor,
+  OauthConfig,
+  OauthParams,
+  OtpParams,
+  RemoveMfaParams,
+  SignupResponse,
+  SubmitOtpCodeResponse,
+  User,
+  VerifyMfaParams,
+} from "./client/types.js";
+import { BaseSignerClient } from "./client/base.js";
+import { SmartAccountSignerFromClient } from "./smartAccountSigner.js";
+
+export interface AlchemyApiKeySignerClientParams {
+  connection: ConnectionConfig;
+  apiKey: {
+    publicKey: string;
+    privateKey: string;
+  };
+}
+
+/**
+ * AlchemyApiKeySignerClient is a client for signing messages using an API key.
+ * It extends the BaseSignerClient and uses the ApiKeyStamper for signing.
+ * Primarily intended to be used server-side.
+ */
+export class AlchemyApiKeySignerClient extends BaseSignerClient<undefined> {
+  /**
+   * Creates an instance of AlchemyApiKeySignerClient.
+   *
+   * @param {AlchemyApiKeySignerClientParams} params The parameters for the client, including the API key and connection config
+   * @param {ConnectionConfig} params.connection The connection configuration for the client
+   * @param {string} params.apiKey.publicKey The public key of the API key
+   * @param {string} params.apiKey.privateKey The private key of the API key
+   */
+  constructor({ connection, apiKey }: AlchemyApiKeySignerClientParams) {
+    const stamper = new ApiKeyStamper({
+      apiPrivateKey: apiKey.privateKey,
+      apiPublicKey: apiKey.publicKey,
+    });
+    super({ connection, stamper });
+  }
+
+  /**
+   * Sets the user for the client
+   *
+   * @param {User} user The user object to set
+   */
+  public setUser = (user: User) => {
+    this.user = user;
+  };
```

Comment:
is this required? base has a `user` with a setter available already

Reply:
i need a way to bubble it up from the `AlchemyApiKeySignerClient` to `createApiKeySigner`, and i'm avoiding doing it within the constructor since it depends on an async call.

Reply:
oh wait, nvm i forgot that `whoami` sets it already.